src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSnapshotManagerDispatcher.cs

```diff
@@ -17,6 +17,9 @@ internal abstract class ProjectSnapshotManagerDispatcher
         public Task RunOnDispatcherThreadAsync(Action action, CancellationToken cancellationToken)
             => Task.Factory.StartNew(action, cancellationToken, TaskCreationOptions.None, DispatcherScheduler);
 
+        public Task RunOnDispatcherThreadAsync<T>(Action<T, CancellationToken> action, T state, CancellationToken cancellationToken)
+            => Task.Factory.StartNew(() => action(state, cancellationToken), cancellationToken, TaskCreationOptions.None, DispatcherScheduler);
```

Comment:
Hmm, given we're still capturing `state` here doesn't that still capture the solution? Did you mean to pass `state` to the task factory?

Reply:
We still capture items in `state`, but not other items. For most of the actions, this means we still capture `NewSolution` but we don't capture `OldSolution`.

Reply:
Ah I see, ok that's what I was missing 😄